README.md

```diff
@@ -433,8 +433,14 @@ authentication
     });
 ```
 
-3. **Retrieve credentials**
- Existing credentials will be returned if they are still valid, otherwise the `refresh_token` will be used to attempt to renew them. If the `expires_in` or both the `access_token` and `id_token` values are missing, the method will throw a `CredentialsManagerException`. The same will happen if the credentials have expired and there's no `refresh_token` available.
+3. **Check credentials existence**
+There are cases were you just want to check if a user session is still valid without the need to use the user credentials right away (i.e. to know if you should present the login screen or the main screen). You could achieve a similar result by calling `getCredentials` and checking the success/failure call, but for convenience we include a `hasCredentials` method that can know in advance if a valid token is available or if it has expired but it could be refreshed, without making an additional network call. The same rules of the `getCredentials` method apply:
```

Comment:
> There are cases were you just want to check if a user session is still valid without the need to use the user credentials right away

I think "without the need to use the user credentials right away" can be omitted. It's simple and the example is more than enough.

> You could achieve a similar result

It's not a similar result, this is not a second option. You haven't mentioned the "better" option yet. But I think you just don't even have to mention the "previous workaround". At least not before the simpler/official way.

> for convenience we include a `hasCredentials` method that can know in advance

that can *let you* know